crates/core_arch/src/x86/bt.rs

```diff
@@ -0,0 +1,105 @@
+#[cfg(test)]
+use stdsimd_test::assert_instr;
+
+/// Return the bit at index `b` of 32-bit integer `a`.
+#[inline]
+#[cfg_attr(test, assert_instr(bt))]
+#[unstable(feature = "bittest", issue = "0")]
+pub unsafe fn _bittest_u32(a: &u32, b: u32) -> u8 {
+    let r: u8;
+    asm!("btl $2, $1\nsetc ${0:b}"
+         : "=r"(r)
+         : "m"(a), "lr"(b)
+         : "cc", "flags", "fpsr");
```

Comment:
clang tags these with `sideeffect` - could you post a godbolt link to this Rust implementation so that we can check that the LLVM-IR is correct ?

Reply:
So I referenced Clang to update the inline assembly I was using. After doing so I'm reasonably confident that the reason Clang is using `sideffect` is because there is an observed write-back to `a` and clang only uses `r` as a constraint. I'm able to produce identical assembly using `=*r` which should indicate that a write-back. 

If I switch back to the inline asm I had before I think this should be valid without the `sideeffect` label (which is good as it doesn't look like rust inline asm supports that).

This is my first time trying to use inline asm like this so I may be totally wrong.

Reply:
> (which is good as it doesn't look like rust inline asm supports that

I think Rust uses `"volatile"` for this, but I wasn't suggesting that the older version was incorrect (clang might also as well just have a bug). I just wanted to compare the implementation with clang at the LLVM-IR level, because precisely what you mention about the inline assembly syntaxes being identical. If the LLVM-IR generated differs, then we can try to reason about why that's the case, and whether the bug is in our side or on clang's side. 

cc @Amanieu

Reply:
Don't use "m" constraints, they are broken. Use "*m" and pass `&a` instead of `a`. "flags" and "fpsr" are not needed here, only "cc" is clobbered. Finally, use plain "r" instead of "lr".